johnzon-core/src/main/java/org/apache/johnzon/core/Snippet.java

```diff
@@ -0,0 +1,435 @@
+                            break;
+                        }
+                        write(keyval.getKey(), keyval.getValue());
+                    }
+                    generator.writeEnd();
+                    generator.flush();
+
+                    break;
+                default: {
+                    generator.write(name, value);
+                    generator.flush();
+                }
+            }
+        }
+
+        private String get() {
+            generator.flush();
+            return snippet.isTruncated() ? snippet.get() + "..." : snippet.get();
+        }
+
+        @Override
+        public void close() {
+            generator.close();
+        }
+    }
+
+    /**
+     * Specialized OutputStream with three internal states:
+     * Writing, Completed, Truncated.
+     *
+     * When there is still space left for more json, the
+     * state will be Writing
+     *
+     * If the last write brought is exactly to the end of
+     * the max length, the state will be Completed.
+     *
+     * If the last write brought us over the max length, the
+     * state will be Truncated.
+     */
+    static class SnippetOutputStream extends OutputStream {
+
+        private final ByteArrayOutputStream buffer;
+        private OutputStream mode;
+
+        public SnippetOutputStream(final int max) {
+            this.buffer = new ByteArrayOutputStream(Math.min(max, 8192));
+            this.mode = new Writing(max, buffer);
+        }
+
+        public String get() {
+            return buffer.toString();
+        }
+
+        /**
+         * Calling this method implies the need to continue
+         * writing and a question on if that is ok.
+         *
+         * It impacts internal state in the same way as
+         * calling a write method.
+         *
+         * @return true if no more writes are possible
+         */
+        public boolean terminate() {
+            if (mode instanceof Truncated) {
+                return true;
+            }
+
+            if (mode instanceof Completed) {
+                mode = new Truncated();
+                return true;
+            }
+
+            return false;
+        }
+
+        public boolean isTruncated() {
+            return mode instanceof Truncated;
+        }
+
+        @Override
+        public void write(final int b) throws IOException {
+            mode.write(b);
+        }
+
+        @Override
+        public void write(final byte[] b) throws IOException {
+            mode.write(b);
+        }
+
+        @Override
+        public void write(final byte[] b, final int off, final int len) throws IOException {
+            mode.write(b, off, len);
+        }
+
+        @Override
+        public void flush() throws IOException {
+            mode.flush();
+        }
+
+        @Override
+        public void close() throws IOException {
+            mode.close();
+        }
+
+        public void print(final String string) {
```

Comment:
unused? let's drop it?

side note: we shouldn't call getBytes() without the encoding anywhere there so since it is the single location doing it, dropping it solves it ;)

Reply:
Dropped!  Was previously used for serializing the constants and forgot to clean that up.  Good catch.